### DISCUSSION TITLE
Unidirectional ManyToMany : rename "id" field

### DISCUSSION THREAD

**Discussion by edgarogh** [doctrine/doctrine2#10004] [Support Questions] [answered]
Hello, I'm trying to create a `ManyToMany` relation between two entities. For the sake of clearness, I'll use the classic "question/tag" example. In my case, I don't need "reverse" indexing to be possible, that is, I just need each question to have a list of tags but I will never index by tag (but if there aren't performance issues, I'm fine with it being there anyways).

The problem is that my `Tag` entity doesn't have an ID as a primary key, but only a string slug named "slug".

Here's the field of my `Question` class :
```php
class Question {
    ...
    #[ORM\ManyToMany(targetEntity: Tag::class, fetch: 'EAGER')]
    private $tags;
    ...
}
```

It looks like Doctrine is looking for a primary key named "id" specifically instead of just using the first primary key. I don't know how to change that. I've played a lot with `indexBy`, `mappedBy`and `inversedBy` with no success. I don't know if I have to put a `#[ManyToMany]` on the tag's primary key too either...

I come from less advanced ORMs and manual SQL, hence my difficulty to grasp high-level concepts like these. Thanks a lot for any help !

**Comment by edgarogh** [comment, marked as answer]
**TLDR Answer :** [`#[InverseJoinColumn(referencedColumnName: 'slug')]`](https://www.doctrine-project.org/projects/doctrine-orm/en/2.13/reference/attributes-reference.html#joincolumn-inversejoincolumn)

---

Well, after spending the last afternoon on this question, I didn't expect to find an answer this quickly. An error message suggested that Doctrine makes a clear distinction between fields (which map to columns) and relations (which don't necessarily have a column, at least not on the same table). So I actually defined a dummy field to act as my reverse relation on the `Tag` end. It looks like this :

```php
    #[ORM\ManyToMany(targetEntity: Question::class, mappedBy: 'tags')]
    #[ORM\JoinColumn(referencedColumnName: 'slug')]
    private $questions;
```

Then, I stumbled upon [this StacOverflow answer](url) with a promising piece of code :

```php
     * @ORM\JoinTable(
     *     name="table1_table2",
     *     joinColumns={
     *          @ORM\JoinColumn(name="foo_uid", referencedColumnName="uniqueIdentifier")
     *     },
     *     inverseJoinColumns={
     *          @ORM\JoinColumn(name="bar_uid", referencedColumnName="uniqueIdentifier")
     *     }
     * )
     */
```

I only needed this `inverseJoinColumn` to be explicit, I don't really care about anything else here. A bit of searching around for an updated attribute-based syntax lead me to the official docs : https://www.doctrine-project.org/projects/doctrine-orm/en/2.13/reference/attributes-reference.html#joincolumn-inversejoincolumn
